As the qemu-kvm version of the i8259 contains KVM bits, it still has to
be compiled per target. This unbreaks migration of the i8259.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---

Not sure if anyone bothers (no one should actually use qemu-kvm for
targets != x86), but let's avoid needless breakages of other targets
requiring the i8259.

 Makefile.objs   |    2 +-
 Makefile.target |    8 ++++----
 hw/i8259.c      |    2 --
 3 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/Makefile.objs b/Makefile.objs
index 13afd19..77237e1 100644
--- a/Makefile.objs
+++ b/Makefile.objs
@@ -223,7 +223,7 @@ hw-obj-$(CONFIG_APPLESMC) += applesmc.o
 hw-obj-$(CONFIG_SMARTCARD) += usb-ccid.o ccid-card-passthru.o
 hw-obj-$(CONFIG_SMARTCARD_NSS) += ccid-card-emulated.o
 hw-obj-$(CONFIG_USB_REDIR) += usb-redir.o
-hw-obj-$(CONFIG_I8259) += i8259.o
+# hw-obj-$(CONFIG_I8259) += i8259.o
 
 # PPC devices
 hw-obj-$(CONFIG_PREP_PCI) += prep_pci.o
diff --git a/Makefile.target b/Makefile.target
index 0b610ad..29eaa68 100644
--- a/Makefile.target
+++ b/Makefile.target
@@ -236,7 +236,7 @@ obj-$(CONFIG_IVSHMEM) += ivshmem.o
 
 # Hardware support
 obj-i386-y += vga.o
-obj-i386-y += mc146818rtc.o pc.o
+obj-i386-y += mc146818rtc.o pc.o i8259.o
 obj-i386-y += cirrus_vga.o sga.o apic.o ioapic.o piix_pci.o
 obj-i386-y += vmport.o
 obj-i386-y += device-hotplug.o pci-hotplug.o smbios.o wdt_ib700.o
@@ -255,7 +255,7 @@ obj-i386-$(CONFIG_KVM_DEVICE_ASSIGNMENT) += device-assignment.o
 obj-ppc-y = ppc.o ppc_booke.o
 obj-ppc-y += vga.o
 # PREP target
-obj-ppc-y += mc146818rtc.o
+obj-ppc-y += mc146818rtc.o i8259.o
 obj-ppc-y += ppc_prep.o
 # OldWorld PowerMac
 obj-ppc-y += ppc_oldworld.o
@@ -311,7 +311,7 @@ obj-mips-y += acpi.o acpi_piix4.o
 obj-mips-y += mips_addr.o mips_timer.o mips_int.o
 obj-mips-y += vga.o
 obj-mips-y += jazz_led.o
-obj-mips-y += gt64xxx.o mc146818rtc.o
+obj-mips-y += gt64xxx.o mc146818rtc.o i8259.o
 obj-mips-y += cirrus_vga.o
 obj-mips-$(CONFIG_FULONG) += bonito.o vt82c686.o mips_fulong2e.o
 
@@ -392,7 +392,7 @@ obj-m68k-y += m68k-semi.o dummy_m68k.o
 
 obj-s390x-y = s390-virtio-bus.o s390-virtio.o
 
-obj-alpha-y = mc146818rtc.o
+obj-alpha-y = mc146818rtc.o i8259.o
 obj-alpha-y += vga.o cirrus_vga.o
 obj-alpha-y += alpha_pci.o alpha_dp264.o alpha_typhoon.o
 
diff --git a/hw/i8259.c b/hw/i8259.c
index fa63e83..a9ea9c9 100644
--- a/hw/i8259.c
+++ b/hw/i8259.c
@@ -697,8 +697,6 @@ static int kvm_kernel_pic_load_from_user(PicState *s)
     return 0;
 }
 
-extern void apic_set_irq_delivered(void);
-
 static void kvm_i8259_set_irq(void *opaque, int irq, int level)
 {
     int pic_ret;
-- 
1.7.3.4
--
To unsubscribe from this list: send the line "unsubscribe kvm" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
